core/templates/pages/skill-editor-page/skill-editor-access.guard.spec.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for SKillEditorAccessGuard.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { ActivatedRouteSnapshot, Router, RouterStateSnapshot, NavigationExtras } from '@angular/router';
+
+import { AppConstants } from 'app.constants';
+import { UserInfo } from 'domain/user/user-info.model';
+import { UserService } from 'services/user.service';
+import { SkillEditorAccessGuard } from './skill-editor-access.guard';
+
+
+class MockRouter {
+  navigate(commands: string[], extras?: NavigationExtras): Promise<boolean> {
+    return Promise.resolve(true);
+  }
+}
+
+describe('Skill editor access guard', () => {
+  let skillEditorAccessGuard: SkillEditorAccessGuard;
+  let userService: UserService;
+  let router: Router;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [UserService,
+        { provide: Router, useClass: MockRouter }, SkillEditorAccessGuard],
+    }).compileComponents();
+
+    skillEditorAccessGuard = TestBed.inject(SkillEditorAccessGuard);
+    userService = TestBed.inject(UserService);
+    router = TestBed.inject(Router);
+  });
+
+  it('should redirect user to 401 page if user is not curriculum admin',
+    async() => {
+      const getUserInfoAsyncSpy = spyOn(
+        userService, 'getUserInfoAsync').and.returnValue(
+        Promise.resolve(UserInfo.createDefault())
+      );
+      const navigateSpy = spyOn(router, 'navigate').and.callThrough();
+
+      const canActivate = await skillEditorAccessGuard.canActivate(
+        {} as ActivatedRouteSnapshot,
+        {} as RouterStateSnapshot
+      );
+
+      expect(canActivate).toBeFalse();
+      expect(getUserInfoAsyncSpy).toHaveBeenCalledTimes(1);
+      expect(navigateSpy).toHaveBeenCalledWith(
+        [`${AppConstants.PAGES_REGISTERED_WITH_FRONTEND.ERROR.ROUTE}/401`]
+      );
+    });
+
+  it('should allow user to access the page if user is curriculum admin',
```

Comment:
This doesn't depend on user role, right? (as it depends on accessValidatorfunction)

Reply:
Done.